src/lib/adapters/waku/safe-waku.ts

```diff
@@ -104,7 +106,7 @@ export class SafeWaku {
 		this.subscriptions = new Map()
 	}
 
-	async sendMessage(message: Message, encryptionKey: Uint8Array, sigPrivKey: Uint8Array) {
+	async sendMessage(message: unknown, encryptionKey: Uint8Array, sigPrivKey: Uint8Array) {
```

Comment:
Why is the message `unknown` now? I would assume that when sending we know the shape, when receiving we don't.

Reply:
Good question! :smile: 

Historically this was originally in the `waku.ts` but I lifted it here, because it had an identical implementation as the `storeDocument` function. Now there is only `storeDocument` at the waku module and this one just calls the `storeEncrypted` which is just a safe wrapper around `storeDocument`.

You are right that here we know the shape, however I did another refactor on the `Message` type. We differentiate between the messages that we send and the messages that we receive. The former does not have metadata and that is what is sent over the wire, the latter has because we augment it with the metadata that we receive from the waku `DecodedMessage` type.  

So this should be a message without metadata type, however I did not manage to express that correctly in Typescript. I just added a commit to indicate that this has the type `WithoutMeta<Message>` but Typescript narrows it to just `{ type: 'user' | 'data' | 'invite' | 'babble' }` which does not provide real type safety. However it is better than `unknown` and at least expresses the intention.